src/unused_finder/mod.rs

```diff
@@ -110,103 +122,269 @@ pub fn find_unused_items(
         .flatten()
         .collect();
 
-    let walked_files_map: HashMap<String, ImportExportInfo> = flattened_walk_file_data
-        .drain(0..)
-        .map(|f| {
-            (
-                no_ext(&f.source_file_path).to_string(),
-                f.import_export_info,
-            )
+    let total_files = flattened_walk_file_data.len();
+    let mut used_files: HashMap<String, &mut WalkFileMetaData> =
+        HashMap::with_capacity(flattened_walk_file_data.len());
+
+    let entry_files: Vec<WalkFileMetaData> = flattened_walk_file_data
+        .iter_mut()
+        .filter_map(|file| {
+            if entry_packages.contains(&file.package_name) {
+                let mut f = file.clone();
+                process_import_export_info(&mut f, &tsconfig);
+                return Some(f);
+            }
+            None
+        })
+        .collect();
+    let mut unused_files: HashMap<String, &mut WalkFileMetaData> = flattened_walk_file_data
+        .iter_mut()
+        .filter_map(|f| {
+            if f.source_file_path == "shared/internal/owa-service/src/contract/TokenResponse.ts" {
+                dbg!(entry_packages.contains(&f.package_name));
+            }
+            if !entry_packages.contains(&f.package_name) {
+                process_import_export_info(f, &tsconfig);
+                return Some((f.source_file_path.clone(), f));
+            }
+            None
         })
         .collect();
-    // HashMap wher key = the used file path, value = a hashset with the items imported from that file, note that those items could not belong to that file (for the cases of export from)
-    let mut path_unused_items_map: HashMap<String, HashSet<ExportedItem>> = HashMap::new();
-
-    walked_files_map
-        .clone()
-        .drain()
-        .for_each(|(path, imp_exp_info)| {
-            // Populate `unused_file_exports`
-            path_unused_items_map.insert(path, imp_exp_info.exported_ids);
-        });
-
-    let mut unused_files: HashSet<_> = walked_files_map.keys().clone().into_iter().collect();
 
-    let resolved_imports_map = get_map_of_imports(&tsconfig, &walked_files_map);
-    resolved_imports_map
-        .iter()
-        .for_each(|(_path, imported_path_items_map)| {
-            // Iterate over each file and mark the imported items as used in the origin file.
-            imported_path_items_map
-                .iter()
-                .for_each(|(imported_path, imported_items)| {
-                    if let Some(origin_file_exported_items) =
-                        path_unused_items_map.get_mut(imported_path)
-                    {
-                        imported_items.iter().for_each(|item| {
+    dbg!(unused_files.len());
+    entry_files.iter().for_each(|file| {
```

Comment:
This initialization step just seems like a single iteration, but with entry_files as the initial set. Is there a difference I'm missing? Because it seems like we could simplify this by merging it with the below looping logic.